src/components/ArticleDetail.jsx

```diff
@@ -1,3 +1,123 @@
+import { useState } from "react";
+import ShowCommentsByArticle from "./ShowCommentsByArticle";
+import { patchData } from "../utilities/API";
+
+const ArticleDetail= ({article}) => {
+
+  const [article_id,setArticleId] = useState("");
+  const [articleVotes,setArticleVotes] = useState(0);
+  const [articleVotesAdd,setArticleVotesAdd] = useState(false);
+  const [articleVotesMinus,setArticleVotesMinus] = useState(false);
+  const [errMsg,setErrMsg] = useState(false)
+
+  const handleFormViewComments = (e) => {
+    e.preventDefault();
+ }
+
+  const HandleOnClickArticleVoteAdd = () => {
+
+    setArticleVotes((currentArticleVotes) => {
+      return  currentArticleVotes + 1;
+    })
+
+    const  newVote = {inc_votes: 1};
+    const articleIdNum = Number(article.article_id);
+    const  patchUrl = `/articles/${articleIdNum}`;
+    setErrMsg(false);
+
+      patchData(patchUrl,newVote)
+       .catch((error) => {          
+           setArticleVotes((currentArticleVotes) => {
+            return  currentArticleVotes - 1;
+           })          
+           setErrMsg(true);
+      })
+
+      if (errMsg === false) {
+         setArticleVotesMinus(false);
+         setArticleVotesAdd(true)
+      }
+    
+  }
```

Comment:
This function and the below function are very similar, can you think of a way to refactor these two functions into one function that can be used for both incrementing a vote and decrementing a vote?